src/test/java/teammates/logic/core/FeedbackResponseCommentsLogicTest.java

```diff
@@ -346,6 +353,288 @@ public void testGetFeedbackResponseCommentsForQuestionInSection_withSectionName_
+        response = dataBundle.feedbackResponses.get("response1ForQ1S1C1");
+        student = dataBundle.students.get("student1InCourse1");
+        studentsInTeam = roster.getTeamToMembersTable().getOrDefault(student.getTeam(), Collections.emptyList());
+        studentsEmailInTeam = studentsInTeam.stream()
+                .map(StudentAttributes::getEmail)
+                .collect(Collectors.toSet());
+
+        ______TS("success: question is visible to instructor; visibility follows feedback question;"
+                + "comment is visible to receiver by default");
+        assertTrue(frcLogic.isResponseCommentVisibleForUser("student1InCourse1@gmail.tmt", UserRole.STUDENT,
+                student, studentsEmailInTeam, response, relatedQuestion, relatedComment));
+
+        ______TS("success: question is visible to instructor; visibility follows feedback question;"
+                + "comment is visible to giver by default");
+        assertTrue(frcLogic.isResponseCommentVisibleForUser("student4InCourse1@gmail.tmt", UserRole.STUDENT,
+                student, studentsEmailInTeam, response, relatedQuestion, relatedComment));
+
+        ______TS("success: question is visible to instructor; visibility follows feedback question;"
+                + "comment is visible to receiver by instructor");
+        assertTrue(frcLogic.isResponseCommentVisibleForUser("instructor1@course1.tmt", UserRole.INSTRUCTOR,
+                student, studentsEmailInTeam, response, relatedQuestion, relatedComment));
+
+        ______TS("failure: question is visible to instructor; visibility follows feedback question;"
+                + "comment is visible to receiver by teammate");
+        assertFalse(frcLogic.isResponseCommentVisibleForUser("instructor3@course1.tmt", UserRole.STUDENT,
+                student, studentsEmailInTeam, response, relatedQuestion, relatedComment));
+
+        relatedQuestion.setShowResponsesTo(Arrays.asList(FeedbackParticipantType.STUDENTS));
+
+        ______TS("failure: question is visible to instructor; visibility follows feedback question;"
+                + "comment is visible to receiver");
+        assertTrue(frcLogic.isResponseCommentVisibleForUser("student1@course1.tmt", UserRole.STUDENT,
+                student, studentsEmailInTeam, response, relatedQuestion, relatedComment));
+
+        ______TS("failure: question is visible to instructor; visibility follows feedback question;"
+                + "comment is visible to teammate");
+        assertTrue(frcLogic.isResponseCommentVisibleForUser("student3@course1.tmt", UserRole.STUDENT,
+                student, studentsEmailInTeam, response, relatedQuestion, relatedComment));
+
+        ______TS("failure: question is visible to instructor; visibility follows feedback question;"
+                + "comment is visible to student not from the same team");
+        assertTrue(frcLogic.isResponseCommentVisibleForUser("student2@course1.tmt", UserRole.STUDENT,
```

Comment:
here

Reply:
Sorry for the careless mistakes. It is fixed.